--- xfig.3.2.3c/w_print.c.orig	Sat Aug  5 14:45:00 2000
+++ xfig.3.2.3c/w_print.c	Sat Aug  5 15:01:20 2000
@@ -958,9 +958,10 @@
 {
     FILE   *printcap;
     char    str[300];
-    int     i,j,len;
+    int     i,j,k,len;
     int     printers;
     Boolean comment;
+    Boolean dudprinter;
 
     if ((printcap=fopen("/etc/printcap","r"))==NULL)
 	return 0;
@@ -993,6 +994,20 @@
 		break;
 	}
 	str[j] = '\0';
+        /* Check for empty printer name or duplicate name */
+        dudprinter = True;
+        for (k=0; k<j; k++) {
+            if(str[k] !=' ' && str[k] != '\t')
+                dudprinter = False;
+        }
+        if(printers > 0) {
+            for (k=0; k<printers; k++) {
+                if(strncmp(names[k],&str[i],j-i+1) == 0)
+                    dudprinter = True;
+            }
+        }
+        if (dudprinter == True)
+            continue;
 	if ((names[printers] = malloc(j-i+1)) == NULL)
 	    return printers;
 	strncpy(names[printers],&str[i],j-i+1);
