diff -urN xfig.3.2.3d.org/Fig.ad xfig.3.2.3d/Fig.ad
--- xfig.3.2.3d.org/Fig.ad	Thu Oct 25 23:14:36 2001
+++ xfig.3.2.3d/Fig.ad	Thu Oct 25 23:15:04 2001
@@ -158,7 +158,7 @@
 
 ! If xfig is compiled with I18N option, you may want to uncomment following line.
 !
-!Fig.international: true
+Fig.international: true
 
 ! Note that you shouldn't remove the following "Fig*international: false".  
 ! It must be there to cancel any *international: true that may defined via xrdb.
diff -urN xfig.3.2.3d.org/Imakefile xfig.3.2.3d/Imakefile
--- xfig.3.2.3d.org/Imakefile	Thu Oct 25 23:14:34 2001
+++ xfig.3.2.3d/Imakefile	Thu Oct 25 23:15:04 2001
@@ -120,7 +120,7 @@
 XCOMM If your setlocale() dosen't support the locale, you should
 XCOMM add -DSETLOCALE to I18N_DEFS.
 
-XCOMM #define I18N
+#define I18N
 
 XCOMM If using an input tablet uncomment the following
 
diff -urN xfig.3.2.3d.org/f_read.c xfig.3.2.3d/f_read.c
--- xfig.3.2.3d.org/f_read.c	Thu Oct 25 23:14:35 2001
+++ xfig.3.2.3d/f_read.c	Thu Oct 25 23:19:59 2001
@@ -15,6 +15,7 @@
  *
  */
 
+#include <locale.h>
 #include "fig.h"
 #include "figx.h"
 #include "resources.h"
@@ -221,7 +222,13 @@
     else {
 	if (!update_figs)
 	    put_msg("Reading objects from \"%s\" ...", file_name);
+#ifdef I18N
+	setlocale(LC_NUMERIC, "C");
+#endif
 	status = readfp_fig(fp, obj, merge, xoff, yoff, settings);
+#ifdef I18N
+	setlocale(LC_NUMERIC, "");
+#endif
 	fclose(fp);
 	/* so subsequent file_msg() calls don't print wrong file name */
 	first_file_msg = False;
diff -urN xfig.3.2.3d.org/f_save.c xfig.3.2.3d/f_save.c
--- xfig.3.2.3d.org/f_save.c	Thu Oct 25 23:14:35 2001
+++ xfig.3.2.3d/f_save.c	Thu Oct 25 23:18:43 2001
@@ -15,6 +15,7 @@
  *
  */
 
+#include <locale.h>
 #include "fig.h"
 #include "resources.h"
 #include "mode.h"
@@ -90,6 +91,9 @@
 
     if (!update_figs)
 	put_msg("Writing . . .");
+#ifdef I18N
+    setlocale(LC_NUMERIC, "C");
+#endif
     write_file_header(fp);
     for (a = objects.arcs; a != NULL; a = a->next) {
 	num_object++;
@@ -115,6 +119,9 @@
 	num_object++;
 	write_text(fp, t);
     }
+#ifdef I18N
+    setlocale(LC_NUMERIC, "");
+#endif
     if (ferror(fp)) {
 	fclose(fp);
 	return (-1);
diff -urN xfig.3.2.3d.org/main.c xfig.3.2.3d/main.c
--- xfig.3.2.3d.org/main.c	Thu Oct 25 23:14:35 2001
+++ xfig.3.2.3d/main.c	Thu Oct 25 23:15:04 2001
@@ -632,7 +632,6 @@
 
 #ifdef I18N
     setlocale(LC_ALL, "");
-    setlocale(LC_NUMERIC, "C");
     XtSetLanguageProc(NULL, NULL, NULL);
 #endif  /* I18N */
 
