--- xfig-3.2.8b/app-defaults/Fig.in.orig	2023-07-01 12:02:17.194898128 +0200
+++ xfig-3.2.8b/app-defaults/Fig.in	2023-07-01 12:02:50.441384683 +0200
@@ -180,9 +180,9 @@ Fig.save8bit:	true
 
 ! If xfig is compiled with I18N option, you may want to uncomment following lines.
 !
-!Fig.international: true
-!Fig.localeEncoding: true
-!Fig.alwaysUseFontSet: true
+Fig.international: true
+Fig.localeEncoding: true
+Fig.alwaysUseFontSet: true
 
 ! Note that you shouldn't remove the following "Fig*international: false".
 ! It must be there to cancel any *international: true that may defined via xrdb.
