diff -urN xfig.3.2.3d.org/Fig.ad xfig.3.2.3d/Fig.ad
--- xfig.3.2.3d.org/Fig.ad	Thu Oct 25 23:14:36 2001
+++ xfig.3.2.3d/Fig.ad	Thu Oct 25 23:15:04 2001
@@ -158,7 +158,7 @@
 
 ! If xfig is compiled with I18N option, you may want to uncomment following line.
 !
-!Fig.international: true
+Fig.international: true
 
 ! Note that you shouldn't remove the following "Fig*international: false".  
 ! It must be there to cancel any *international: true that may defined via xrdb.
