diff -Nru xfig.3.2.3c/Fig.ad xfig.3.2.3c.new/Fig.ad
--- xfig.3.2.3c/Fig.ad	Tue Jul 25 19:05:57 2000
+++ xfig.3.2.3c.new/Fig.ad	Sat Nov  4 12:26:07 2000
@@ -16,7 +16,7 @@
 
 ! pdfviewer - put your favorite pdf viewer here.  
 !		This is for viewing the xfig how-to guide and man pages
-Fig.pdfviewer:			acroread %f
+Fig.pdfviewer:			xpdf %f
 
 ! Spell check program - put your favorite spelling check program here.
 !		It must write the misspelled words to standard output.
--- xfig.3.2.4/Imakefile.orig	Thu Dec 19 18:57:11 2002
+++ xfig.3.2.4/Imakefile	Tue Mar 25 02:05:31 2003
@@ -49,7 +49,7 @@
 XCOMM are in different places
 
 PNGLIBDIR = $(USRLIBDIR)
-PNGINC = -I/usr/local/include
+PNGINC =
 
 XCOMM If don't want JPEG support, comment out the #define USEJPEG line
 XCOMM Uncomment the #define for USEJPEG if you want to be able to import
@@ -68,8 +68,8 @@
 
 #ifdef USEJPEG
 #ifdef USEINSTALLEDJPEG
-JPEGLIBDIR = /usr/local/lib
-JPEGINC = -I/usr/include/X11
+JPEGLIBDIR = /usr/lib
+JPEGINC =
 #else
 JPEGLIBDIR = ../jpeg
 JPEGINC = -I$(JPEGLIBDIR)
@@ -92,7 +92,7 @@
 
 #ifdef USEXPM
 XPMLIBDIR = $(USRLIBDIR)
-XPMINC = -I/usr/include/X11
+XPMINC = -I/usr/X11R6/include/X11
 #endif
 
 XCOMM Uncomment the following definiton if you want to use the small icons
@@ -154,7 +154,7 @@
 XCOMM inline functions. With the "INLINE" keyword, you should notice that
 XCOMM the display will be a bit faster in complex figures
 
-XCOMM USEINLINE = -DUSE_INLINE
+USEINLINE = -DUSE_INLINE
 
 XCOMM use (and change) the following if you want the multi-key data base file
 XCOMM somewhere other than the standard X11 library directory
@@ -234,7 +234,7 @@
 #endif /* USEJPEG */
 
 #ifdef I18N
-I18N_DEFS = -DI18N -DSETLOCALE
+I18N_DEFS = -DI18N
 I18N_SRC = w_i18n.c
 I18N_OBJ = w_i18n.o
 #endif
